### ISSUE TITLE
Add whitebalance colour picker

### ISSUE THREAD

**Issue by masc4ii** [ilia3101/MLV-App#26]
> @masc4ii I don't think it's a good idea to do it fully in Qt, as it doesn't have access to linear non saturated values, so the internals will have to be somewhere within the raw/processing libs. But if you can sort out a colour picker solution, that would be good.

I agree. My idea is:
- I create a picker in Qt, which only gets a (x,y) coordinate in the frame(number)
- the non-Qt and raw processing part offers a function `wbPicker(frameNr, x, y)` (or something in the way) and this one does the calculation
- afterwards we get the new temperature and tint value and set the sliders (which will send the new values again to the libs) and in the end the new picture is shown

@ilia3101 : What do you think?

**Comment by ilia3101** [comment]
Yeas, I will try and figure out a way to implement this in the processing and raw thing. I have some ideas - main thing is to have x and y coordinates for where.

Also will take an average with a radius of maybe 8 pixels, that should lead to a better white balance result.

**Comment by masc4ii** [comment]
I'll copy it here again to not forget where you wrote it:
> I'm thinking it will work something like
setMlvProcessingWhiteBalance(mlvObject_t * video, int x, int y, uint64_t frame);
(after some minor thought experiments)
EDIT: could also have a parameter int pickerSize (or summin like that)

**Comment by masc4ii** [comment]
Qt side should work so far...
What I did:
- Picker symbol in symbol bar & menu
- Activating it will change cursor icon on graphics view
- Clicking anywhere in the frame changes icon back to normal and unchecks picker symbol
- x & y is sended from graphicsview to main class

**Comment by ilia3101** [comment]
goood progress!
I'll begin to do this after cache is commited to the cache branch tommorow.

**Comment by bouncyball-git** [comment]
@ilia3101: How's the back end progress?

**Comment by bouncyball-git** [comment]
@ilia3101: 
This is very out of topic there. Here is the more appropriate place:

[Link1](https://github.com/ilia3101/MLV-App/commit/42f84beb09645a7835a7f58124cd4a6b1d4b0485#commitcomment-25057404)
[Link2](https://github.com/ilia3101/MLV-App/commit/42f84beb09645a7835a7f58124cd4a6b1d4b0485#commitcomment-25057576)

**Comment by masc4ii** [comment]
Hej @ilia3101 , any progress here? Will it be possible to implement this on your side? I am really missing this functionality. I was playing with Lightroom this afternoon, and I was using wbPicker soooo often... :) pleeeeaaaase (:

**Comment by bouncyball-git** [comment]
Pleeeeeeeease :)

**Comment by SoyLeor** [comment]
Hello, how can I get the temperature correction done by exporting in cDNG as MLRV does?

**Comment by masc4ii** [comment]
cDNG is RAW. So there is no temperature correction possible. If you need a corrected file, chose TIFF or ProRes or...

**Comment by bouncyball-git** [comment]
Manual WB correction for DNG is possible, needs calculating of "AsShotNeutral" tag. I just did not implement it yet. Had no motivation w/o working picker ;)

**Comment by masc4ii** [comment]
So you overwrite the original tag value then?!

**Comment by bouncyball-git** [comment]
Ye it can be definitely done.

**Comment by masc4ii** [comment]
It feels a bit like christmas+birthday+easter+... 
🎉  THE WHITEBALANCE PICKER IS WORKING!!! 🎉 

Try it with this commit:
https://github.com/ilia3101/MLV-App/commit/8a6ddef84e80faa44588465aa2a948cb9906d25a

My algorithm is very stupid and easy, that's why it is very very slow: I try each (reduced, that we don't have to wait minutes) possible value combination of slider settings and look where processed color is the greyest. So there is space to make it better... maybe some quick-search-algorithm could speed it up?!

**Comment by bouncyball-git** [comment]
I have almost no time to have fun with MLV App these days... damn it.

**Comment by masc4ii** [comment]
What's going on? To much work?

**Comment by bouncyball-git** [comment]
Yup. Some evenings I don't wanna look at the display any more :)

**Comment by masc4ii** [comment]
I could now boost the speed of the wb search algorithm: the matrix was calculated always for the whole 16bit range, for each possible temperature and tint value combination - but we need it just for the RGB which was clicked by the picker. So now the matrix is (while finding algortihm is active) only for these R, G and B values. The speed boost is so huge, that I could raise precision of temp & tint by 10 and it is still faster than before... yeah.
